### PR TITLE
update shap dependency to 0.33.0

### PR DESCRIPTION
**Pull request by imatiach-msft**
update shap dependency to 0.33.0
also some minor cleanup in setup.py:
- removed unused logic for major/patch version
- add python 3.7 classifier
- reference github url which should give interpret-community a github statistics panel on the pypi webpage

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import shutil
 
 _major = '0.3'
```
Comment (wamartin-aml) [inline python/setup.py]: I still think this should be major and minor

File: python/setup.py [hunk 2]
```diff
@@ -54,7 +42,7 @@
     'scikit-learn',
     'packaging',
     'interpret-core[required]==0.1.19',
-    'shap>=0.20.0, <=0.32.1'
+    'shap>=0.20.0, <=0.33.0'
```
Comment (wamartin-aml) [inline python/setup.py]: Does this kind of change necessitate a bump to 0.4.0? I'd like to get more clarity on when we patch and when we bump minor version

Comment (imatiach-msft) [reply python/setup.py]: I always do a separate PR for the version bump to make it clear in the commit history

File: python/setup.py (hunk 1 above)
Comment (imatiach-msft) [reply python/setup.py]: done!

**Pull request merged**